promql/test.go

```diff
@@ -51,6 +52,15 @@ const (
 
 var testStartTime = time.Unix(0, 0).UTC()
 
+//go:embed testdata
+var TestManifestFS embed.FS // TestManifestFS contains a set of tests.
```

Comment:
Hm  I am not sure I am fan of this. Can you even import such embedded FS with Go Modules?

I think Go modules already copies all test files - you access them through you `GOMODCACHE`, so why not that?

![image](https://user-images.githubusercontent.com/6950331/222724975-95c2d08b-0d2e-4b98-adbc-1db5c11c45de.png)

Essentially it might add more complexity to build system then needed and it's not consistent with other test data we have in Prometheus. What about prometheus-engine using GOMODCACHE as the first step? (Other changes in this PRs are valid).

Reply:
I hacked this into prometheus-engine and imported through that embedded FS without issues though.

Reply:
Still, the inconsistency with other testdata in Prometheus module (10+ of those?) is a bigger issue.

Reply:
I could add a `RunBuiltinTests` method that handles this to the `Test` struct, how does that sound? That way this does not need to be exported.

Reply:
Or maybe that's fine - cc @roidelapluie for opinion (:

Reply:
Added a `RunBuiltinTests` method but its implementation still needs embedded test manifests, but they are not exported anymore, is that acceptable?